pkg/asset/cluster/azure/azure.go

```diff
@@ -33,12 +34,14 @@ func PreTerraform(ctx context.Context, clusterID string, installConfig *installc
 	if len(installConfig.Config.Azure.ResourceGroupName) == 0 {
```

Comment:
probably need to change this too, because if users are not using an existing resource group, which is unrelated to byo vnets, this function will return early. I think it makes sense to just move this check within the `tagResourceGroup` func

Reply:
ah yes I was unsure of how this should work.
the one thing about this is that to tag the vnet we need a resource group name. where can we get it from at this point?

Reply:
Seems I've been using the wrong resource group name to search for the vnet (thanks QE for catching it), fixed it to use the network resource group name: https://github.com/openshift/installer/pull/7611/commits/c6f8c4977f89268d458f757bd56fd7c7045455bc#diff-80d6651bbc295bfb8b7cab8e213625e2f8b2335a2a64a2a36bb2f2973201b1f9R55-R58
I also added a fall back to the resource group name but I don't know whether it's necessary.
PTAL